server/datastore/mysql/software_titles.go

```diff
@@ -410,6 +411,13 @@ GROUP BY st.id, package_self_service, package_name, package_version, package_url
 		defaultFilter += ` AND ( si.self_service = 1 OR vat.self_service = 1 ) `
 	}
 
+	// if excluding fleet maintained apps, join on the fleet_library_apps table by bundle ID
+	// and filter out any row from software_titles that has a matching row in fleet_library_apps.
+	if opt.ExcludeFleetMaintainedApps {
+		softwareJoin += ` LEFT JOIN fleet_library_apps fla ON st.bundle_identifier = fla.bundle_identifier`
+		additionalWhere += " AND fla.id is NULL"
```

Comment:
Just because this table is so, it might be worth trying a `WHERE NOT EXISTS` instead of an anti-join. https://erikdarling.com/why-not-exists-makes-more-sense-than-left-joins/

Reply:
That's excellent!  Moved to WHERE NOT EXISTS :+1: